spine_json_lib/spine_animation_editor.py

```diff
@@ -171,7 +171,18 @@ def clean_animation(self):
         print("Removed slots {}".format(slots_to_remove))
 
         self.remove_attachments(attachments_ids=attachments_to_remove)
-        print("Removed attachments {}".format(attachments_to_remove))
+        detached_attachments = self.spine_graph.remove_heads_of_type(
+            NodeType.ATTACHMENT.name
+        )
+        detached_attachments += self.spine_graph.remove_leafs_of_type(
+            NodeType.ATTACHMENT.name
+        )
+        print(
+            "Removed attachments {}".format(
+                list(attachments_to_remove) + detached_attachments
+            )
+        )
```

Comment:
this print and others should be added as part of a logger with different levels... it's out of the scope of this PR, but should be an improvement